src/main/java/com/jcraft/jsch/PortWatcher.java

```diff
@@ -211,7 +212,9 @@ public void run(){
           channel.connect(connectTimeout);
         } else {
           ChannelDirectTCPIP channel = new ChannelDirectTCPIP();
+          channel.setSession(session);
           channel.init();
+          channel.setSession(session);
```

Comment:
Why do we have to call `setSession()` twice here (both before and after `init()`?

Reply:
Shouldn't be. I'll remove the second one (I suppose while going over the changes I've missed the first setSession I've already added and added the second one).